src/derive/stages/batches.rs

```diff
@@ -281,3 +281,88 @@ impl Batch {
+
+        let data = hex::decode(data).unwrap();
+        let txs = vec![data.clone()];
+        let (tx, rx) = mpsc::channel();
+        let batch_transactions = BatcherTransactions::new(rx);
+
+        let res = tx.send(BatcherTransactionMessage {
+            txs,
+            l1_origin: 123456,
+        });
+        assert!(res.is_ok());
+
+        let config = Config {
+            l1_rpc_url: "".to_string(),
+            l2_rpc_url: "".to_string(),
+            l2_engine_url: "".to_string(),
+            chain: ChainConfig::base_goerli(),
+            jwt_secret: "".to_string(),
+            rpc_port: 0,
+        };
+        let channels = Channels::new(batch_transactions, Arc::new(config.clone()));
+
+        let state = State::new(
+            BlockInfo {
+                hash: Default::default(),
+                number: 0,
+                parent_hash: Default::default(),
+                timestamp: 0,
+            },
+            Epoch {
+                number: 0,
+                hash: Default::default(),
+                timestamp: 0,
+            },
+            Arc::new(config.clone()),
+        );
+        let mut batches = Batches::new(
+            channels,
+            Arc::new(RwLock::new(state)),
+            Arc::new(config.clone()),
+        );
+        assert!(batches.next().is_none());
```

Comment:
Shouldn't we be constructing this test to try and get a batch from this?

Reply:
@ncitron yeah, just draft, will change this assertion.